pkg/client/cloudevents/grpcsource/watcherstore.go

```diff
@@ -283,6 +283,14 @@ func (m *RESTFulAPIWatcherStore) registerWatcher(namespace string, labelSelector
 
 	watcher, ok := m.watchers[namespace]
 	if ok {
+		if watcher.isStopped() {
```

Comment:
can we make it simpler?

```golang
if ok {
    if !watcher.isStopped {
        return watcher
    }
}
```
I do not think we need to explicily delete it from the map, it will be override in line 291